src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java

```diff
@@ -61,19 +67,68 @@ public ApiResponse<Void> invokeChsKafkaApi(String contextId, String companyNumbe
     }
 
     private ChangedResource mapChangedResource(String contextId, String companyNumber,
-                                               String chargeId) {
-        String resourceUri = String.format(COMPANY_CHARGES_URI, companyNumber, chargeId);
-
+                                               String chargeId, ChargeApi chargeApi) {
         ChangedResourceEvent event = new ChangedResourceEvent();
         event.setType(CHANGED_EVENT_TYPE);
         event.setPublishedAt(String.valueOf(OffsetDateTime.now()));
         ChangedResource changedResource = new ChangedResource();
-        changedResource.setResourceUri(resourceUri);
+        if (chargeApi != null) {
+            changedResource.setDeletedData(chargeApi);
+            event.setType(DELETE_EVENT_TYPE);
+        }
+
+        changedResource.setResourceUri(String.format(
+                COMPANY_CHARGES_URI, companyNumber, chargeId));
         changedResource.event(event);
         changedResource.setResourceKind(resourceKind);
         changedResource.setContextId(contextId);
 
         return changedResource;
     }
 
+    private PrivateChangedResourcePost getChangedResourcePost(String uri,
+                                                              ChangedResource changedResource) {
+
+        InternalApiClient internalApiClient = apiClientServiceImpl.getInternalApiClient();
```

Comment:
Nice to have: Maybe create InternalApiClient Spring bean like we did in charges-delta-consumer and autowire instead?

Reply:
This is an existing code part - i just moved it to a new method to facilitate and reuse it for the delete functionality. if you want this to be improvised i can do it part of unhappy path http status codes story (DSND-781).

Reply:
sure, lets do it in 781